server/service/src/stocktake/insert.rs

```diff
@@ -110,6 +119,90 @@ pub fn insert_stocktake(
     Ok(result)
 }
 
+fn insert_rows(
+    input: InsertStocktake,
+    connection: &StorageConnection,
+    ctx: &ServiceContext,
+    stocktake_id: &str,
+) -> Result<(), InsertStocktakeError> {
+    let mut item_ids = Vec::<String>::new();
+    // check for master list
+    if let Some(master_list_id) = input.master_list_id {
+        let master_list_items =
+            MasterListLineRepository::new(&connection).query_by_filter(MasterListLineFilter {
+                master_list_id: Some(EqualFilter::equal_to(&master_list_id)),
+                id: None,
+                item_id: None,
+            })?;
+        master_list_items.iter().for_each(|line| {
+            item_ids.push(line.item_id.clone());
+        });
+    }
+    // insert a stocktake row for each item & stock_line
+    item_ids.iter().for_each(|item_id| {
+        let stock_lines = StockLineRepository::new(&connection)
+            .query_by_filter(
+                StockLineFilter::new().item_id(EqualFilter::equal_to(item_id)),
+                Some(ctx.store_id.clone()),
+            )
+            .unwrap();
+
+        if stock_lines.len() == 0 {
+            insert_stocktake_line(&connection, stocktake_id, &item_id, None);
+        } else {
+            stock_lines.iter().for_each(|stock_line| {
+                insert_stocktake_line(&connection, stocktake_id, &item_id, Some(stock_line));
+            });
+        }
+    });
+    Ok(())
+}
+
+fn insert_stocktake_line(
+    connection: &StorageConnection,
+    stocktake_id: &str,
+    item_id: &String,
+    stock_line: Option<&StockLine>,
+) {
+    let mut stock_line_id = "None".to_string();
+    let mut stocktake_line_row = StocktakeLineRow {
+        id: uuid(),
+        stocktake_id: stocktake_id.to_string(),
+        location_id: None,
+        comment: None,
+        snapshot_number_of_packs: 0.0,
+        counted_number_of_packs: None,
+        item_id: item_id.to_string(),
+        note: None,
+        inventory_adjustment_reason_id: None,
+        stock_line_id: None,
+        batch: None,
+        expiry_date: None,
+        pack_size: None,
+        cost_price_per_pack: None,
+        sell_price_per_pack: None,
+    };
+    if let Some(stock_line) = stock_line {
+        let stock_line_row = stock_line.stock_line_row.clone();
```

Comment:
I like destructuring to make sure things are not missed, like `location_id` (initially and in the future when/if new fields are added), even though it's more vertical code, it's generally better imo, also I think this is a use case for splitting the method into two. As an example, together with previous comments (just cause it seemed fun to do):

https://github.com/openmsupply/open-msupply/compare/938-create-stocktake...938-example